### ISSUE TITLE
Featured image is not shared with posts on Facebook

### ISSUE THREAD

**Issue by vytisbulkevicius** [Codeinwp/tweet-old-post#997]
### Description

This looks to be happening for a while, but when a post is shared on Facebook, the featured image of the post is missing and not shared.

It's not related to recent updates but probably something what Facebook changed in the past.

Featured image is there set for the post:
![image](https://github.com/user-attachments/assets/2fec2a39-b2df-41d7-a896-8b3d48c6d03c)

Logs:
![image](https://github.com/user-attachments/assets/894bfad2-113a-472c-8c56-2087b27d2d70)

Post on Facebook:
![image](https://github.com/user-attachments/assets/64bd3b6b-8c31-4dfc-bc9c-61e3808855b7)

### Step-by-step reproduction instructions

1. Connect your ROP to Facebok account
2. Have posts with featured images that can be shared (or create new one)
3. Enable sharing and wait for the post to be shared
4. Check shared post on Facebook - it doesn't have an image

### Is the issue you are reporting a regression

No

**Comment by AndreeaCristinaRadacina** [comment]
+1 https://wordpress.org/support/topic/facebook-no-featured-images/

**Comment by girishpanchal30** [comment]
Hey @vytisbulkevicius,

I've enabled the `Share As Image Post` setting and it seems to share the post with an image on Facebook.
https://tinyurl.com/22ry55za
https://tinyurl.com/2dgnm2wo

Social media platforms capture the post image automatically from the `og:image` meta tag if the above setting is disabled.
Ref: https://muffingroup.com/blog/facebook-link-thumbnail-not-working/#:~:text=Why%20is%20my%20Facebook%20link,to%20refresh%20the%20link%20preview.

Thanks

**Comment by vytisbulkevicius** [comment]
Hi @girishpanchal30,

Thanks for checking. But `Share As Image Post` is a different thing - https://docs.revive.social/article/958-how-to-share-posts-as-image-posts-to-social-accounts

It was possible to share a post with text preview and also image. Yes, social platforms look for og:image when you just share URL manually but with ROP as we're sharing post from the dashboard we should be looking for attachment somewhere in the code when sharing and using that attachment as image for the post shared. It was working before this way but something was changed how FB accept the parameters I assume.

**Comment by girishpanchal30** [comment]
@vytisbulkevicius Okay, I'll debug the code and try to find out the issue.

**Comment by girishpanchal30** [comment]
@vytisbulkevicius I've reviewed the code and I think `article` sharing gets the featured image automatically. I do not see any code related to simple article post sharing with images.

Here:
https://github.com/Codeinwp/tweet-old-post/blob/8f4688b3dab6cb90eae09609788899ddf7fd3eb7/includes/admin/services/class-rop-facebook-service.php#L565-L577

I've checked with the live site and it fetch the image automatically.

Ref: https://tinyurl.com/2buccfmu

**Comment by selul** [comment]
@vytisbulkevicius how is the Share image post different from what you had in mind? I see that Share image post is basically sharing the image with text, which is what you were looking to have?

**Comment by vytisbulkevicius** [comment]
@selul @girishpanchal30,

So the different between these two are these:

**Image Post**
With this method (and when feature is enabled in ROP) post is shared as an image and where the text is posted, there is title and then the URL added, it looks like this:
<img width="658" alt="image" src="https://github.com/user-attachments/assets/2e298048-0c6e-4037-ba0c-bc0fb5bbded2">

In this method, if you click on the image on Facebook, you get image opened in gallery:
<img width="1477" alt="image" src="https://github.com/user-attachments/assets/49e26eea-aed5-4bcb-80c6-107ceacc1a4b">

Regular sharing (Text Post):

In this case, image is also fetched but clicking on the image leads you to the article (destination website).
It looks like this (it's the same post):
<img width="631" alt="image" src="https://github.com/user-attachments/assets/61adfc60-7a5b-4e0e-bdfe-1201863255a2">


Here is also a short guide that covers this difference: https://help.socialpilot.co/article/504-what-is-the-difference-between-image-post-text-post-in-feed-content

I think it's important for the 2nd scenario to work as well as it's more common to want users to open your website clicking the image rather than opening that image in the gallery.

Also, it's interesting that with this post this time it worked for me when testing and image was shared but with another post (and other image used) it did not work and we can see that there are some users complaining about this functionality.

**Comment by selul** [comment]
@vytisbulkevicius the 2nd scenario is relying on OG tags from what I know, @girishpanchal30 not sure if that controlled from the API call instead of relying on OG tags? 

For links that don't have an OG tag we can use https://developers.facebook.com/tools/debug/ to debug why this is happening.

**Comment by girishpanchal30** [comment]
Hey @vytisbulkevicius,

The 2nd scenario is sharing `Text Post` which means facebook generates a link preview automatically. I think we can handle this kind of thing from API.

I used the debug tool to test a few of our live post URLs, but I could not see any problems.

For more details: https://www.blog2social.com/en/faq/content/4/131/en/social-media-post-formats-_-the-differences-between-image-post-and-link-post.html

Thanks

**Comment by vytisbulkevicius** [comment]
Thank you. I'm closing this then, I thought we have control over it. 

I tested those different posts from my test instance and with FB debug tool I also get image in only one of them for some reason (can't spot different in how they are created).

I tried to replace the image on the post where it was not working and then it picked the image so it looks to be about the image itself and maybe the dimensions, size or something else  (they are both JPG) of it because it still shows that `og:image` is not defined but still picks it with another image:

<img width="1017" alt="image" src="https://github.com/user-attachments/assets/794ceaf3-5617-4ece-a855-9c80dbd55b07">
